roles/nova-common/defaults/main.yml

```diff
@@ -23,7 +27,7 @@ nova:
   python_libvirt_version: 1.2.2-0ubuntu2~cloud0
   qemu_kvm_version: 2.0.0+dfsg-2ubuntu1.16~cloud0
   librdb1_version: 0.80.9-0ubuntu0.14.04.2~cloud0
-  glance_endpoint: http://{{ endpoints.glance }}:9292
+  glance_endpoint: "{{ endpoints.glance.url.internal }}"
```

Comment:
good point. So would your recommendation be that keystone endpoints are all same, but nova.conf (if any other place needed) is wired ti http://ucarp_ip:9393 ?